js/bybit.js

```diff
@@ -2356,54 +2374,120 @@ module.exports = class bybit extends Exchange {
+        //        "time":"1652115973053",
+        //        "updateTime":"1652115973063",
+        //        "isWorking":true
+        //     }
+        //
+        // create order usdc
+        //      {
+        //            "orderId":"34450a59-325e-4296-8af0-63c7c524ae33",
+        //            "orderLinkId":"",
+        //            "mmp":false,
+        //            "symbol":"BTCPERP",
+        //            "orderType":"Limit",
+        //            "side":"Buy",
+        //            "orderQty":"0.00100000",
+        //            "orderPrice":"20000.00",
+        //            "iv":"0",
+        //            "timeInForce":"GoodTillCancel",
+        //            "orderStatus":"Created",
+        //            "createdAt":"1652261746007873",
+        //            "basePrice":"0.00",
+        //            "triggerPrice":"0.00",
+        //            "takeProfit":"0.00",
+        //            "stopLoss":"0.00",
+        //            "slTriggerBy":"UNKNOWN",
+        //            "tpTriggerBy":"UNKNOWN"
         //     }
         //
         const marketId = this.safeString (order, 'symbol');
         market = this.safeMarket (marketId, market);
         const symbol = market['symbol'];
         let feeCurrency = undefined;
-        const timestamp = this.parse8601 (this.safeString2 (order, 'created_at', 'created_time'));
-        const id = this.safeString2 (order, 'order_id', 'stop_order_id');
-        const type = this.safeStringLower (order, 'order_type');
-        let price = undefined;
-        if (type !== 'market') {
-            price = this.safeString (order, 'price');
-        }
-        const average = this.safeString (order, 'average_price');
-        const amount = this.safeString (order, 'qty');
+        let timestamp = this.parse8601 (this.safeString2 (order, 'created_at', 'created_time'));
+        if (timestamp === undefined) {
+            timestamp = this.safeNumber2 (order, 'time', 'transactTime');
+            if (timestamp === undefined) {
+                timestamp = this.safeIntegerProduct (order, 'createdAt', 0.001);
+            }
+        }
+        const id = this.safeStringN (order, [ 'order_id', 'stop_order_id', 'orderId' ]);
+        const type = this.safeStringLowerN (order, [ 'order_type', 'type', 'orderType' ]);
+        const price = this.safeString2 (order, 'price', 'orderPrice');
+        const average = this.safeString2 (order, 'average_price', 'avgPrice');
+        const amount = this.safeStringN (order, [ 'qty', 'origQty', 'orderQty' ]);
         const cost = this.safeString (order, 'cum_exec_value');
-        const filled = this.safeString (order, 'cum_exec_qty');
+        const filled = this.safeString2 (order, 'cum_exec_qty', 'executedQty');
         const remaining = this.safeString (order, 'leaves_qty');
         const marketTypes = this.safeValue (this.options, 'marketTypes', {});
         const marketType = this.safeString (marketTypes, symbol);
```

Comment:
This is the only place where we use this outdated dictionary of `marketTypes`. It was introduced in order to address the market differences with the old endpoints, but we don't really need it anymore, since we can now determine if a market is linear or not from the market structure itself. I suggest that we remove `marketTypes` entirely.

Reply:
The end-goal is to perform this switch: https://github.com/ccxt/ccxt/pull/13263/files#diff-cb8b530c783df8bccc6e0df910e7433b34c9c9c3cea8904fcf2c59f1e60aeebeR645

Reply:
@kroitor agreed and removed!